src/tdastro/opsim/opsim.py

```diff
@@ -463,9 +491,40 @@ def range_search(self, query_ra, query_dec, radius=None):
 
         # Adjust the angular radius to a cartesian search radius and perform the search.
         adjusted_radius = 2.0 * np.sin(0.5 * np.radians(radius))
-        return self._kd_tree.query_ball_point(cart_query, adjusted_radius)
-
-    def get_observations(self, query_ra, query_dec, radius=None, cols=None):
+        inds = self._kd_tree.query_ball_point(cart_query, adjusted_radius)
+
+        if t_min is not None or t_max is not None:
+            num_queries = len(query_ra)
+            times = self.table[self.colmap["time"]].to_numpy()
+
+            if t_min is None:
+                t_min = np.full(num_queries, -np.inf)
+            elif np.isscalar(t_min):
+                t_min = np.full(num_queries, t_min)
+            if len(t_min) != num_queries:
+                raise ValueError(f"t_min must be a scalar or an array of length {num_queries}.")
+
+            if t_max is None:
+                t_max = np.full(num_queries, np.inf)
+            elif np.isscalar(t_max):
+                t_max = np.full(num_queries, t_max)
+            if len(t_max) != num_queries:
+                raise ValueError(f"t_max must be a scalar or an array of length {num_queries}.")
+
+            # Run through each list of indices and filter by time. We need to do this
+            # iteratively, because the lists can have different lengths.
+            for idx, subinds in enumerate(inds):
+                if len(subinds) == 0:
+                    continue
+                time_mask = (times[subinds] >= t_min[idx]) & (times[subinds] <= t_max[idx])
+                inds[idx] = np.asarray(subinds)[time_mask]
+
+        # If the query was a scalar, we return a single list of indices.
+        if ra_scalar or dec_scalar:
```

Comment:
Should it be `and`? Should we also check t_min/max to be scalars?
```suggestion
        if ra_scalar and dec_scalar:
```